docs/architecture/adr-074-timeout-params.md

```diff
@@ -0,0 +1,228 @@
+on the fastest hardware with a summed voting power of +2/3 of the network's total have
+completed execution of the block. Allowing the network to proceed as soon as the fastest
++2/3 completed execution would have a cumulative effect over heights, eventually
+leaving slower validators unable to participate in consensus at all. `TimeoutCommit`
+therefore allows networks to have greater variability in hardware. Additional
+discussion of this can be found in [tendermint issue 5911][tendermint-issue-5911-comment]
+and [spec issue 359][spec-359].
+
+## Alternative Approaches
+
+### Hardcode the parameters
+
+Many Tendermint networks run on similar cloud-hosted infrastructure. Therefore,
+they have similar bandwidth and machine resources. The timings for propagating votes
+and blocks are likely to be reasonably similar across networks. As a result, the
+timeout parameters are good candidates for being hardcoded. Hardcoding the timeouts
+in Tendermint would mean entirely removing these parameters from any configuration
+that could be altered by either an application or a node operator. Instead,
+Tendermint would ship with a set of timeouts and all applications using Tendermint
+would use this exact same set of values.
+
+While Tendermint nodes often run with similar bandwidth and on similar cloud-hosted
+machines, there are enough points of variability to make configuring
+consensus timeouts meaningful. Namely, Tendermint network topologies are likely to be
+very different from chain to chain. Additionally, applications may vary greatly in 
+how long the `Commit` phase may take. Applications that perform more work during `Commit`
+require a longer `TimeoutCommit` to allow the application to complete its work
+and be prepared for the next height.
+
+* From https://github.com/tendermint/spec/issues/359 it looks like `TimeoutCommit` really only exists for the purpose of allowing
+precommits to be collected after the +2/3 has initially been seen. Not much more value to it than that. 
+
+* allowing long timeout commit discussed here: https://github.com/tendermint/tendermint/issues/5911#issuecomment-973560381
+
+## Decision
+
+None
+
+## Detailed Design
+
+### New Consensus Parameters
+
+A new `TimeoutParams` `message` will be added to the [params.proto file][consensus-params-proto].
+This message will have the following form:
+
+```proto
+message TimeoutParams {
+ google.protobuf.Duration timeout_propose = 1
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ google.protobuf.Duration timeout_propose_delta = 2
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ google.protobuf.Duration timeout_vote = 3
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ google.protobuf.Duration timeout_vote_delta = 4
+       [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+ google.protobuf.Duration timeout_commit = 5
```

Comment:
I wonder if having the `timeout` prefix to all these variables causes it to stutter i.e. `params.timeout.timeout_propose`

Reply:
done.